DSL/com.larsreimann.safeds/src/main/kotlin/com/larsreimann/safeds/staticAnalysis/linking/ArgumentToParameter.kt

```diff
@@ -45,3 +48,37 @@ fun SdsArgumentList.parametersOrNull(): List<SdsParameter>? {
         else -> null
     }
 }
+
+/**
+ * Returns the [SdsParameter] that corresponds to this [SdsGoalArgument] or `null` if it cannot be resolved.
+ */
+fun SdsGoalArgument.parameterOrNull(): SdsParameter? {
+    return when {
+        isNamed() -> parameter.asResolvedOrNull()
+        else -> {
+            val goalArgumentList = closestAncestorOrNull<SdsGoalArgumentList>() ?: return null
+            val parameters = goalArgumentList.parametersOrNull() ?: return null
+            val lastParameter = parameters.lastOrNull()
+
+            val firstNamedArgumentIndex = goalArgumentList.arguments.indexOfFirst { it.isNamed() }
+            val thisIndex = goalArgumentList.arguments.indexOf(this)
+
+            return when {
+                lastParameter?.isVariadic == true && thisIndex >= parameters.size - 1 -> lastParameter
+                firstNamedArgumentIndex != -1 && thisIndex > firstNamedArgumentIndex -> null
+                else -> parameters.getOrNull(thisIndex)
+            }
+        }
+    }
+}
+
+/**
+ * Returns the list of [SdsParameter]s that corresponds to this list of [SdsGoalArgument]s or `null` if it cannot be
+ * resolved.
+ */
+fun SdsGoalArgumentList.parametersOrNull(): List<SdsParameter>? {
+    return when (val parent = this.eContainer()) {
+        is SdsGoalCall -> parent.parametersOrNull()
+        else -> null
+    }
+}
```

Comment:
Should be in a separate file

Reply:
Also: Is there a way to do without all this duplication? Maybe make `SdsArgument` a supertype of `SdsGoalArgument` and `SdsArgumentList` a supertype of `SdsGoalArgumentList`?  Or is the behavior of them somehow different?